### PR TITLE
Login and signup backend done

### PR DESCRIPTION
**Pull request by lakshyabatman**
Login and Signup backend done.
Although the client side is facing CORS issue. @capedcrusader23

### CODE REVIEW

**Comment by lakshyabatman** [comment]
@capedcrusader23 , sir the proxy table is situated in /client  as vue.config.js and Login request is send from Components/Login.vue, you can see CORS error in console :)

**Comment by lakshyabatman** [comment]
I addition, Project schema is added, addProject functionality is added and everything is integrated with frontend :D

**Comment by lakshyabatman** [comment]
In additon, Project schema is added, add Project functionality is added and everything is integrated with frontend.
I found out why my commites were not squashed cause i added comments Between commits, so they are merged in my local machine. But not here :)

**Comment by lakshyabatman** [comment]
@jaskirat2000 @vaibhavdaren @capedcrusader23 @devshiva619, please review and merge :D

File: client/public/service-worker.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+if (workbox) {
+  console.log(`Workbox is loaded`)
+
+  workbox.precaching.precacheAndRoute(self.__precacheManifest)
+  console.log('HI HI GHI HIH IHIHIH')
```
Comment (vaibhavdaren) [inline client/public/service-worker.js]: change to  meaningful logs

File: client/src/components/Signup.vue [hunk 2]
```diff
@@ -139,7 +139,7 @@
                                     class="mb-2 mr-sm-2 mb-sm-0"
                                     :value="null"
                                     v-model="form.gender"
-                                    :options="{ 'male': 'Male', 'female': 'Female','other':'Other' }"
+                                    :options="{ 0: 'Male', 1: 'Female',2:'Other' }"
```
Comment (vaibhavdaren) [inline client/src/components/Signup.vue]: use enum

File: server/controller/authcontroller.js [hunk 3]
```diff
@@ -1,74 +1,117 @@
...
-                {
-                    const payload={id:us._id};
-                    const tok=await jwt.sign(payload,secret,{expiresIn:3600})
-                      res.json({
-                        sucess:true,
-                        token:'Bearer '+tok
-                      })
-                }
-            } 
+        if(!isValid) return res.status(400).json({error, status:0});
```
Comment (vaibhavdaren) [inline server/controller/authcontroller.js]: search on best practices to use status codes.

File: server/controller/profile.controller.js [hunk 4]
```diff
@@ -17,6 +18,166 @@ const { check, validationResult } = require('express-validator/check');
...
+        await user.findByIdAndUpdate(SubscribingBy,{$push: {'followingList' : SubscribingTo }})
+      }
+      catch(err) {
+        console.log(err)
+        return res.send({status:0,msg:'Some Error Occured'})
+      }
+    } 
+    console.log('Subscribing User UPDATED')
+    // Now Le'ts update the subscribed user
+    // NOTE: MAKE THIS ASYNC/AWAIT
```
Comment (vaibhavdaren) [inline server/controller/profile.controller.js]: TO DO:

File: client/src/components/Signup.vue (hunk 2 above)
Comment (lakshyabatman) [reply client/src/components/Signup.vue]: Enums in template in vuejs?

File: server/controller/authcontroller.js (hunk 3 above)
Comment (lakshyabatman) [reply server/controller/authcontroller.js]: I have used 400 for bad request and 401 for auth error.

File: server/controller/profile.controller.js (hunk 4 above)
Comment (lakshyabatman) [reply server/controller/profile.controller.js]: I'll complete it by today.

**Pull request merged**